packages/web-components/fast-foundation/src/data-grid/data-grid-row.ts

```diff
@@ -150,6 +152,20 @@ export class FASTDataGridRow extends FASTElement {
     public cellElements: HTMLElement[];
 
     private behaviorOrchestrator: ViewBehaviorOrchestrator | null = null;
+    /**
+     * If the row is selected.
+     *
+     * @internal
+     */
+    @observable
+    public selected: boolean;
+
+    /**
+     * Whether click select is enabled
+     *
+     * @internal
+     */
+    public disableClickSelect: boolean = false;
```

Comment:
I wonder if this can be delegated to the grid somehow, rather than set on each row. Keeping things in sync like this is somewhat confusing, and if I'm reading correctly below has introduced a bug in that this value is checked for click selection but _not_ for keyboard selection. Perhaps there's a way to move the logic to the grid through a callback instead of an event. This type of thing seems to come up a lot and I'm not sure if there's a good pattern for it.

Reply:
This property is marked as "@internal", meaning the parent grid sets it on the child rows based on the setting on the parent grid.  Authors don't need to worry about it. Replaced with "selectionBehavior" too.